src/extensions/default/InlineColorEditor/main.js

```diff
@@ -59,7 +67,45 @@ define(function (require, exports, module) {
         } while (match && (pos.ch < start || pos.ch > end));
 
         if (!match) {
-            return null;
+            // Check if the cursorLine has a CSS rule of type color
+            var cssPropertyName, semiColonPos, colonPos, colorValue, cursorLineSubstring, firstCharacterPos;
+
+            // Get the css property name after removing spaces and ":" so that we can check for it in the file ColorProperties.json
+            cssPropertyName = cursorLine.split(':')[0].trim();
+
+            if (!cssPropertyName || !properties[cssPropertyName]) {
+                return null;
+            }
+
+            if (properties[cssPropertyName].type === "color") {
+                colonPos = cursorLine.indexOf(":");
+                semiColonPos = cursorLine.indexOf(";");
+                if (semiColonPos !== -1) {
+                    // edit the color value of an existing css rule
+                    cursorLineSubstring = cursorLine.substring(colonPos + 1, semiColonPos);
+                    colorValue = cursorLineSubstring.trim();
+                    if (colorRegEx.test(colorValue)) {
+                        firstCharacterPos = cursorLineSubstring.search(/\S/);
+                        pos.ch = colonPos + 1 + firstCharacterPos;
+                        endPos = {line: pos.line, ch: semiColonPos};
+                    } else {
+                        return null;
+                    }
+                } else {
+                    // edit the color value of a new css rule
+                    pos.ch = colonPos + 1;
```

Comment:
The issue is that we need to have a space after the colon when the user types a new css rule. Currently the status is that if I'm writing a new CSS rule, I can't preserve the space that comes by default after the colon. The marker replaces that space with a color value. Instead we need something way to add another space to that line and then replace that space with the color value because the marker doesn't work for a position, it needs a string to replace.
Current - `color:red;` and we need `color: red`.